generators/chipyard/src/main/resources/csrc/spiketile.cc

```diff
@@ -0,0 +1,917 @@
+                           long long int* dcache_c_data_3,
+                           long long int* dcache_c_data_4,
+                           long long int* dcache_c_data_5,
+                           long long int* dcache_c_data_6,
+                           long long int* dcache_c_data_7,
+
+                           unsigned char dcache_d_valid,
+                           unsigned char dcache_d_has_data,
+                           unsigned char dcache_d_grantack,
+                           long long int dcache_d_sourceid,
+                           long long int dcache_d_data_0,
+                           long long int dcache_d_data_1,
+                           long long int dcache_d_data_2,
+                           long long int dcache_d_data_3,
+                           long long int dcache_d_data_4,
+                           long long int dcache_d_data_5,
+                           long long int dcache_d_data_6,
+                           long long int dcache_d_data_7,
+
+                           unsigned char mmio_a_ready,
+                           unsigned char* mmio_a_valid,
+                           long long int* mmio_a_address,
+                           long long int* mmio_a_data,
+                           unsigned char* mmio_a_store,
+                           int* mmio_a_size,
+
+                           unsigned char mmio_d_valid,
+                           long long int mmio_d_data
+                           )
+{
+  if (!host) {
+    host = context_t::current();
+    sout.rdbuf(std::cerr.rdbuf());
+    log_file = new log_file_t(nullptr);
+  }
+  if (tiles.find(hartid) == tiles.end()) {
+    printf("Constructing spike processor_t\n");
+    isa_parser_t *isa_parser = new isa_parser_t(isa, "MSU");
+    chipyard_simif_t* simif = new chipyard_simif_t(icache_ways, icache_sets,
+                                                   dcache_ways, dcache_sets,
+                                                   cacheable, uncacheable, readonly_uncacheable, executable,
+                                                   icache_sourceids, dcache_sourceids);
+    processor_t* p = new processor_t(isa_parser,
+                                     "vlen:128,elen:64",
+                                     simif,
+                                     hartid,
+                                     false,
+                                     endianness_little,
+                                     false,
+                                     log_file->get(),
+                                     sout);
+
+    p->enable_log_commits();
+
+    s_vpi_vlog_info vinfo;
+    if (!vpi_get_vlog_info(&vinfo))
+      abort();
+    for (int i = 1; i < vinfo.argc; i++) {
+      std::string arg(vinfo.argv[i]);
+      if (arg == "+spike-debug") {
+        p->set_debug(true);
+      }
+      if (arg == "+spike-stq") {
+        simif->use_stq = true;
+      }
+    }
+
+    p->reset();
```

Comment:
Does this comply with the chipyard reset behabior? Is there some reset tree?

Reply:
You are right I'll modify this to behave correctly with reset.